taipy/rest/config/rest_config.py

```diff
@@ -8,84 +8,133 @@
+
 
+class RestConfig(UniqueSection):
+    """Configuration parameters for running the `Rest^` service"""
 
-class RestConfig(Section):
     name: str = "REST"
 
-    def __init__(self):
-        super().__init__("rest")
-        self._port: int = 5000
-        self._host: str = "127.0.0.1"
-        self._use_https: bool = False
-        self._ssl_cert: Optional[str] = None
-        self._ssl_key: Optional[str] = None
-
-    def __copy__(self):
-        new_instance = RestConfig()
-        new_instance._port = self._port
-        new_instance._host = self._host
-        new_instance._use_https = self._use_https
-        new_instance._ssl_cert = self._ssl_cert
-        new_instance._ssl_key = self._ssl_key
-        return new_instance
+    _DEFAULT_PORT: int = 5000
+    _DEFAULT_HOST: str = "127.0.0.1"
+    _DEFAULT_USE_HTTPS: bool = False
+    _DEFAULT_SSL_CERT: Optional[str] = None
+    _DEFAULT_SSL_KEY: Optional[str] = None
 
-    @classmethod
-    def _clean(cls):
-        return RestConfig()
+    def __init__(
+        self,
+        port: int = _DEFAULT_PORT,
+        host: str = _DEFAULT_HOST,
+        use_https: bool = _DEFAULT_USE_HTTPS,
+        ssl_cert: Optional[str] = _DEFAULT_SSL_CERT,
+        ssl_key: Optional[str] = _DEFAULT_SSL_KEY,
+        **properties,
+    ):
+        self._port = port
+        self._host = host
+        self._use_https = use_https
+        self._ssl_cert = ssl_cert
+        self._ssl_key = ssl_key
+        super().__init__(**properties)
+
+    def __copy__(self) -> "RestConfig":
+        return RestConfig(
+            self.port,
+            self.host,
+            self.use_https,
+            self.ssl_cert,
+            self.ssl_key,
+            **copy(self._properties),
+        )
+
+    def _clean(self):
+        self.port = self._DEFAULT_PORT
+        self.host = self._DEFAULT_HOST
+        self.use_https = self._DEFAULT_USE_HTTPS
+        self.ssl_cert = self._DEFAULT_SSL_CERT
+        self.ssl_key = self._DEFAULT_SSL_KEY
+
+    def _update(self, config_as_dict: Dict, default_section=None):
+        self.port = config_as_dict.pop("port", self.port)
+        self.host = config_as_dict.pop("host", self.host)
+        self.use_https = config_as_dict.pop("use_https", self.use_https)
+        self.ssl_cert = config_as_dict.pop("ssl_cert", self.ssl_cert)
+        self.ssl_key = config_as_dict.pop("ssl_key", self.ssl_key)
+
+        self._properties.update(config_as_dict)
+
+    def _to_dict(self):
+        return {
+            key: value
+            for key, value in {
+                "port": self.port,
+                "host": self.host,
+                "use_https": self.use_https,
+                "ssl_cert": self.ssl_cert,
+            }.items()
+            if value is not None
+        }
 
     @classmethod
-    def _to_dict(cls):
-        return {"port": cls.port, "host": cls.host, "use_https": cls.use_https, "ssl_cert": cls.ssl_cert}
+    def _from_dict(cls, data: Dict):
```

Comment:
To make the type checker happy.
```suggestion
    def _from_dict(cls, data: Dict[str, Any], id=None, config = None):
```